File: bot.lua
-- bot.lua
local discordia = require('discordia')
local client = discordia.Client()

-- گرفتن توکن از متغیر محیطی
local token = os.getenv("DISCORD_TOKEN")

client:on('ready', function()
    print('Logged in as '.. client.user.username)
end)

client:on('messageCreate', function(message)
    if message.content == '!ping' then
        message.channel:send('Pong!')
    end
end)

client:run('Bot ' .. token)

File: bot.py
import os
import discord

intents = discord.Intents.default()
client = discord.Client(intents=intents)

@client.event
async def on_ready():
    print(f'We have logged in as {client.user}')

@client.event
async def on_message(message):
    if message.content.startswith('$hello'):
        await message.channel.send('Hello!')

# استفاده از توکن ذخیره شده در GitHub Secrets
client.run(os.getenv('DISCORD_TOKEN'))
